tests/test_objective_funs.py

```diff
@@ -2458,6 +2460,15 @@ def test_objective_no_nangrad_omnigenity(self, helicity):
         g = obj.grad(obj.x())
         assert not np.any(np.isnan(g)), str(helicity)
 
+    @pytest.mark.unit
+    def test_objective_no_nangrad_ballooning(self):
```

Comment:
i don't think this needs a special case, you can just let the regular parameterized test take care of it

Reply:
but you probably will need to special case the `TestComputeScalarResolution` one

Reply:
It does need a special case. If I don't create a special function, I get an error.

Reply:
what error? This is the base case that would be called otherwise:
```python
        eq = Equilibrium(L=2, M=2, N=2)
        obj = ObjectiveFunction(objective(eq), use_jit=False)
        obj.build()
        g = obj.grad(obj.x(eq))
        assert not np.any(np.isnan(g)), str(objective)
```
the only difference I see is the equilibrium, but that shouldn't affect having nans in the gradient?

Reply:
That is the cause. Dario and Rahul asked me this last month. The default equilibrium gives nan on objectives that do coordinate mapping onto field lines. The default `Equilibrium` is an asymmetric circle with no toroidal current, and that affects the rotational transform and hence field line coordinate mapping. Probably a simple nan leak that could be caught with jax's nan debugger.

Reply:
I think this is resolved. @f0uriest run the code after commenting the function to see the error. Basically, some of the gradients become nans.

Reply:
PRobably expected physically given 0 iota